### PR TITLE
feature/IVYPORTAL-16690-Quick-Search-in-Dashboard-widget-LTS-Change-r…

### PR DESCRIPTION
**Pull request by lmluat-axonivy**
…equests

- Add tooltip on search box of the quick search in widget to show search scope
- Add icon infor at below of quick search header in column management

### CODE REVIEW

File: AxonIvyPortal/portal/cms/cms_en.yaml [hunk 1]
```diff
@@ -797,7 +797,9 @@ ch.ivy.addon.portalkit.ui.jsf:
     NoData: No data
     NoSavedFilterMessage: No saved filter found. Try to create a new by click on Save Filter button
     QuickSearch: Quick Search
+    QuickSearchCaution: Be mindful of performance when adding numerous columns.
     QuickSearchInfo: You can customize the Quick Search's scope in the widgets column management. It searches based on "contains," but be mindful of performance when adding numerous columns.
+    QuickSearchScope: 'Search Scope: '
```
Comment (lndanh-axonivy) [inline AxonIvyPortal/portal/cms/cms_en.yaml:802]: We could put the param here:
Search Scope: {0}
And we will pass this paramerter from Java code

File: AxonIvyPortal/portal/src/ch/ivy/addon/portal/generic/bean/DashboardBean.java [hunk 2]
```diff
@@ -433,4 +435,54 @@ public void setImportedDashboards(List<Dashboard> importedDashboards) {
...
+  private String getSearchScopeFromCaseWidget(String searchScope, CaseDashboardWidget caseWidget) {
+    List<String> fieldList = caseWidget.getFilterableColumns().stream()
+        .filter(col -> col.getQuickSearch() != null && col.getQuickSearch().equals(Boolean.TRUE))
+        .map(ColumnModel::getHeaderText).collect(Collectors.toList());
+    StringBuilder fieldNameList = appendFieldNameList(fieldList);
+    searchScope = searchScope + fieldNameList.toString();
+    return searchScope;
+  }
+
+  private StringBuilder appendFieldNameList(List<String> fieldList) {
```
Comment (lndanh-axonivy) [inline AxonIvyPortal/portal/src/ch/ivy/addon/portal/generic/bean/DashboardBean.java:470]: we can write it simpler
  
  private StringBuilder appendFieldNameList(List<String> fieldList) {
	    StringBuilder fieldNameList = new StringBuilder();
	    if (!fieldList.isEmpty()) {
	        fieldNameList.append(String.join(", ", fieldList));
	    } else {
	        fieldNameList.append("none");
	    }
	    return fieldNameList;
	}

File: AxonIvyPortal/portal/src/ch/ivy/addon/portal/generic/bean/DashboardBean.java [hunk 3]
```diff
@@ -433,4 +435,54 @@ public void setImportedDashboards(List<Dashboard> importedDashboards) {
...
+  private String getSearchScopeFromTaskWidget(String searchScope, TaskDashboardWidget taskWidget) {
+    List<String> fieldList = taskWidget.getFilterableColumns().stream()
+        .filter(col -> col.getQuickSearch() != null && col.getQuickSearch().equals(Boolean.TRUE))
+        .map(ColumnModel::getHeaderText).collect(Collectors.toList());
+    StringBuilder fieldNameList = appendFieldNameList(fieldList);
+    searchScope = searchScope + fieldNameList.toString();
+    return searchScope;
+  }
+
+  private String getSearchScopeFromCaseWidget(String searchScope, CaseDashboardWidget caseWidget) {
```
Comment (lndanh-axonivy) [inline AxonIvyPortal/portal/src/ch/ivy/addon/portal/generic/bean/DashboardBean.java:461]: Another way to write this function
  private String getSearchScopeFromWidget(List<ColumnModel> filterableColumns) {
	    List<String> fieldList = filterableColumns.stream()
	        .filter(col -> Boolean.TRUE.equals(col.getQuickSearch()))
	        .map(ColumnModel::getHeaderText)
	        .collect(Collectors.toList());
	    StringBuilder fieldNameList = appendFieldNameList(fieldList);
	    return Ivy.cms().co("/ch.ivy.addon.portalkit.ui.jsf/dashboard/QuickSearchScope", Arrays.asList(fieldNameList.toString()));
	}

File: AxonIvyPortal/portal/src/ch/ivy/addon/portal/generic/bean/DashboardBean.java [hunk 4]
```diff
@@ -433,4 +435,54 @@ public void setImportedDashboards(List<Dashboard> importedDashboards) {
...
+    if (widget instanceof TaskDashboardWidget) {
+      TaskDashboardWidget taskWidget = (TaskDashboardWidget) widget;
+      this.searchScope = getSearchScopeFromTaskWidget(searchScope, taskWidget);
+    }
+    
+    if (widget instanceof CaseDashboardWidget) {
+      CaseDashboardWidget caseWidget = (CaseDashboardWidget) widget;
+      this.searchScope = getSearchScopeFromCaseWidget(searchScope, caseWidget);
+    }
+  }
```
Comment (lndanh-axonivy) [inline AxonIvyPortal/portal/src/ch/ivy/addon/portal/generic/bean/DashboardBean.java:450]: another way to write this function
  public void setSearchScope(DashboardWidget widget) {
    
    if (widget instanceof TaskDashboardWidget taskWidget) {
      this.searchScope = getSearchScopeFromWidget(taskWidget.getFilterableColumns());
    }
    
    if (widget instanceof CaseDashboardWidget caseWidget) {
      this.searchScope = getSearchScopeFromWidget(caseWidget.getFilterableColumns());
    }
  }

File: AxonIvyPortal/portal/src_hd/ch/ivy/addon/portal/generic/dashboard/component/ColumnManagement/ColumnManagement.xhtml [hunk 5]
```diff
@@ -33,7 +33,14 @@
           <p:column headerText="#{ivy.cms.co('/ch.ivy.addon.portalkit.ui.jsf/common/displayName')}">
           #{column.headerText}
         </p:column>
-        <p:column headerText="#{ivy.cms.co('/ch.ivy.addon.portalkit.ui.jsf/dashboard/QuickSearch')}" width="90" styleClass="u-text-align-center">
+        <p:column id="quick-search-header" headerText="#{ivy.cms.co('/ch.ivy.addon.portalkit.ui.jsf/dashboard/QuickSearch')}" width="120" styleClass="u-text-align-center">
+          <f:facet name="header">
+          <h:panelGroup layout="block">
+            <span>#{ivy.cms.co('/ch.ivy.addon.portalkit.ui.jsf/dashboard/QuickSearch')}</span>
+            <h:outputText id="quick-search-infor-icon" styleClass="si si-information-circle" style="padding-left: 5%"/>
```
Comment (lndanh-axonivy) [inline AxonIvyPortal/portal/src_hd/ch/ivy/addon/portal/generic/dashboard/component/ColumnManagement/ColumnManagement.xhtml:40]: we should put it to a css class: style="padding-left: 5%"

**Review by lndanh-axonivy — COMMENTED** [review]
@lmluat-axonivy  please check the review

File: AxonIvyPortal/portal/src_hd/ch/ivy/addon/portal/generic/dashboard/component/ColumnManagement/ColumnManagement.xhtml (hunk 5 above)
Comment (lmluat-axonivy) [reply AxonIvyPortal/portal/src_hd/ch/ivy/addon/portal/generic/dashboard/component/ColumnManagement/ColumnManagement.xhtml]: @lndanh-axonivy  I updated code, please check

Comment (lndanh-axonivy) [reply AxonIvyPortal/portal/src_hd/ch/ivy/addon/portal/generic/dashboard/component/ColumnManagement/ColumnManagement.xhtml]: LGTM, please merge!

**Pull request merged**